app/src/ui/history/compare.tsx

```diff
@@ -338,20 +339,25 @@ export class CompareSidebar extends React.Component<
     const key = event.key
 
     if (key === 'Enter') {
-      if (this.state.filterText === '') {
+      if (this.state.filterText.length === 0) {
         this.handleEscape()
       } else {
-        const branch =
-          this.props.repositoryState.branchesState.allBranches.find(
-            branch =>
-              branch.name.toLowerCase() === this.state.filterText.toLowerCase()
-          ) || null
-
-        this.props.dispatcher.loadCompareState(
-          this.props.repository,
-          DisplayHistory
-        )
-        this.setState({ selectedBranch: branch })
+        if (this.state.selectedBranch == null) {
```

Comment:
Can we call `clearFilterState` here or does the call to `setState` in that method make it not possible?

Reply:
I wasn't sure, so I left it as-is.

Reply:
I think it should be fine either way, so I went ahead and merged it.